# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: layout/generic/nsTextFrame.h
# Commit: 5ded58486aba
# Full Hash: 5ded58486abab5712f305a88bbdd746ae74dcdef
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2023-03-06 09:45:20
# Description:
#   Bug 1819957 - Split out nsTextPaintStyle into its own source file. r=emilio
#   
#   This lets us instantiate it on the stack in nsDisplayText::Paint.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D171635
# ==============================================================================

diff -r 58235f400c11 -r 5ded58486aba layout/generic/nsTextFrame.h
--- a/layout/generic/nsTextFrame.h	Sun Mar 05 20:37:07 2023 +0000
+++ b/layout/generic/nsTextFrame.h	Sun Mar 05 22:24:10 2023 +0000
@@ -8,9 +8,9 @@
 #define nsTextFrame_h__
 
 #include "mozilla/Attributes.h"
-#include "mozilla/gfx/2D.h"
 #include "mozilla/UniquePtr.h"
 #include "mozilla/dom/Text.h"
+#include "mozilla/gfx/2D.h"
 
 #include "nsIFrame.h"
 #include "nsISelectionController.h"
@@ -928,12 +928,11 @@
   // This sets *aShadows to the appropriate shadows, if any, for the given
   // type of selection.
   // If text-shadow was not specified, *aShadows is left untouched.
-  // If aTextPaintStyle is provided, it will be used to avoid the need to
-  // instantiate a temporary one.
+  // Note that the returned shadow(s) will only be valid as long as the
+  // textPaintStyle remains in scope.
   void GetSelectionTextShadow(
-      SelectionType aSelectionType,
-      mozilla::Span<const mozilla::StyleSimpleShadow>* aShadows,
-      nsTextPaintStyle* aTextPaintStyle = nullptr);
+      SelectionType aSelectionType, nsTextPaintStyle& aTextPaintStyle,
+      mozilla::Span<const mozilla::StyleSimpleShadow>* aShadows);
 
   /**
    * Utility methods to paint selection.